test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -1800,6 +1800,39 @@ TEST_F(LuaHttpFilterTest, GetRequestedServerName) {
   EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, true));
 }
 
+// Verify that binary values could also be extracted from dynamicMetadata() in LUA filter.
+TEST_F(LuaHttpFilterTest, GetDynamicMetadataBinaryData) {
+  const std::string SCRIPT{R"EOF(
+    function envoy_on_request(request_handle)
+      local metadata = request_handle:streamInfo():dynamicMetadata():get("envoy.pp")
+      local bin_data = metadata["bin_data"]
+      local str_table = { }
+      local data_length = string.len(metadata["bin_data"])
+      for idx = 1, data_length do
+        if string.byte(bin_data, idx) ~= 0 then
```

Comment:
Why skip the null? Now I think this code isn't really distinguishing whether the 0x00 made it to the lua context or not. Imagine a bug where we instead pass this string through MessageToJson which will drop the 0x00 but keep the rest of the characters.

I like what you had earlier with converting this to hex escape codes since it shows to the reader byte by byte what made it over. Alternatively you could call base64Escape and in your assertion show that the value logged will match calling base64Escape on buffer in your test, but that seems slightly more difficult for the reader to follow.

Reply:
Changed it to hex instead. Thanks!